client/src/main/scripts/components/VideoResultPreview.tsx

```diff
@@ -35,42 +35,77 @@ interface VideoResultPreviewProps {
-    }
+const VideoResultPreview: React.FC<VideoResultPreviewProps> = (props) => {
+  const { searchTerm, sortBy, turns, videoFact } = props;
+  let allVideoResults: JSX.Element[] = [];
+  turns.forEach((turn) => {
+    const turnContent = getTurnContent(turn, videoFact);
+    const multiHighlights = getMultiHighlights(searchTerm, sortBy, turnContent);
+    const videoResults = multiHighlights.map((multiHighlight) => {
+      return (
+        <VideoResult
+          key={getUniqueKey(turn, multiHighlight)}
+          multiHighlight={multiHighlight}
+          onPlayClick={props.eventHandlers.onPlayClick}
+          sortBy={props.sortBy}
+          turn={turn}
+          turnContent={turnContent}
+          videoFact={props.videoFact}
+        />
+      );
+    });
+    allVideoResults = allVideoResults.concat(videoResults);
+  });
+  return allVideoResults.length > 0 ? (
+    <div className="results__preview">
+      <h2 className="results__subheading">
+        {videoFact.fact.title} - {videoFact.fact.primaryDate}
+      </h2>
+      {allVideoResults}
+    </div>
+  ) : null;
+};
+
+const getTurnContent = (
+  turn: number,
+  videoFact: Foundation.VideoFactContent
+): string => {
+  let fullTurnText;
+  const firstWord = videoFact.speakerWord[turn];
+  const firstChar = videoFact.charOffsets[firstWord];
+
+  if (videoFact.speakerWord[turn + 1]) {
+    const lastWord = videoFact.speakerWord[turn + 1];
+    const lastChar = videoFact.charOffsets[lastWord] - 1;
+    fullTurnText = videoFact.plainText.substring(firstChar, lastChar);
+  } else {
+    // Result is in last turn
+    fullTurnText = videoFact.plainText.substring(firstChar);
   }
-  render() {
-    const { searchTerm, sortBy, turns, videoFact } = this.props;
-    return (
-      <div className="results__preview">
-        <h2 className="results__subheading">
-          {videoFact.fact.title} - {videoFact.fact.primaryDate}
-        </h2>
-        {turns.map((turn, idx) => {
-          return (
-            <VideoResultTurnList
-              key={idx.toString()}
-              onPlayClick={this.props.eventHandlers.onPlayClick}
-              searchTerm={searchTerm}
-              sortBy={sortBy}
-              turn={turn}
-              videoFact={videoFact}
-            />
-          );
-        })}
-      </div>
-    );
+  return fullTurnText;
+};
+
+const getMultiHighlights = (
+  searchTerm: string,
+  sortBy: SelectionOptions,
+  turnContent: string
+): MultiHighlight[] => {
+  const turnFinder = new TurnFinder(searchTerm);
+  const turnWithResults = turnFinder.findResults(turnContent);
+  if (sortBy === "Containing") {
+    return turnWithResults.expandBy(1);
+  } else if (sortBy === "BeforeAndAfter") {
+    return turnWithResults.expandBy(2);
+  } else {
+    const msg = "VideoResultTurn: Unknown radio selection";
+    alertErr(msg);
+    throw msg;
   }
-}
+};
+
+const getUniqueKey = (turn: number, { cut }: MultiHighlight): string => {
+  return `${turn}.${cut[0]}.${cut[1]}`;
+};
```

Comment:
Should we add the video title or hash or youtubeId to this?  Collisions where two videos have the same turn and cuts is pretty unlikely, but if we added other documents into the search (say founding documents) and you did a search for the first sentence of each, it's not crazy to think we might get one.

Reply:
As long as the keys are unique within a given list there won't be any issues, and there is a separate list for each video. But if we ever decide to flatten that list, it'll be good to harden this, so I've added the youtube Id.